cloudigrade/internal/metrics.py

```diff
@@ -0,0 +1,220 @@
+"""Celery Metrics."""
+import logging
+import re
+import threading
+import time
+
+from django.utils.translation import gettext as _
+from prometheus_client import Counter, Gauge, Histogram
+
+from config.celery import app as celery_app
+
+
+logger = logging.getLogger(__name__)
+
+
+class CeleryMetrics:
+    """Celery Metrics class for processing and exposing Prometheus metrics."""
+
+    def __init__(self, interval=2, buckets=None):
+        """Celery Metrics Initializer."""
+        self.listener_started = False
+        self.interval = interval
+        self.state_counters = {
```

Comment:
To continue the conversation started at https://github.com/cloudigrade/cloudigrade/pull/1128/files#r826076794

It looks like large chunks of this file were copied directly from celery-exporter's [`exporter.py`](https://github.com/danihodovic/celery-exporter/blob/master/src/exporter.py) without many changes. I _suspect_ that [their BSD license](https://github.com/danihodovic/celery-exporter/blob/master/LICENSE.md) is compatible enough with our generic GPL license that this is okay, but IANAL. 🤷 

What are the key differences between this version and theirs? If the changes aren't substantial, would it make sense to contribute to their project and wait (or use a fork) until they cut a release? Or pull their existing version in as a dependency and make only local overrides/changes where necessary? If not, no worries; I'm just curious about where the differences begin and end.

Also, I'm a little anxious about copying code into our repo that we didn't write or at least fully understand. My original question in that thread was:

> I'm not familiar with these names, aside from the obvious inline strings that describe them. Where did you find them to include here?

We should be able to answer that confidently before taking ownership of this code. Would you give us a summary of how this code works?

Reply:
👍 I'll setup a meeting to go over this. Thanks.